rotkehlchen/tests/unit/test_globaldb.py

```diff
@@ -431,3 +435,133 @@ def test_globaldb_pragma_foreign_keys(globaldb):
     cursor.execute('PRAGMA foreign_keys')
     # Now the pragma should be off
     assert cursor.fetchone()[0] == 0
+
+
+def test_global_db_restore(globaldb, database):
+    """
+    Check that the user can recreate assets information from the packaged
+    database with rotki. The test adds a new asset, restores the database
+    and checks that the added token is not in there and that the amount of
+    assets is the expected
+    """
+    # Add a custom eth token
+    address_to_delete = make_ethereum_address()
+    token_to_delete = EthereumToken.initialize(
+        address=address_to_delete,
+        decimals=18,
+        name='willdell',
+        symbol='DELME',
+    )
+    globaldb.add_asset(
+        asset_id='DELMEID1',
+        asset_type=AssetType.ETHEREUM_TOKEN,
+        data=token_to_delete,
+    )
+    # Add a token with underlaying token
+    with_underlying_address = make_ethereum_address()
+    with_underlying = EthereumToken.initialize(
+        address=with_underlying_address,
+        decimals=18,
+        name="Not a scam",
+        symbol="NSCM",
+        started=0,
+        underlying_tokens=[UnderlyingToken(
+            address=address_to_delete,
+            weight=1,
+        )],
+    )
+    globaldb.add_asset(
+        asset_id='xDELMEID1',
+        asset_type=AssetType.ETHEREUM_TOKEN,
+        data=with_underlying,
+    )
+    # Add asset that is not a token
+    globaldb.add_asset(
+        asset_id='1',
+        asset_type=AssetType.OWN_CHAIN,
+        data={
+            'name': 'Lolcoin',
+            'symbol': 'LOLZ',
+            'started': 0,
+        },
+    )
+
+    # Add asset that is not a token
+    globaldb.add_asset(
+        asset_id='2',
+        asset_type=AssetType.OWN_CHAIN,
+        data={
+            'name': 'Lolcoin2',
+            'symbol': 'LOLZ2',
+            'started': 0,
+        },
+    )
+
+    database.add_asset_identifiers('1')
+    database.add_asset_identifiers('2')
+
+    # Try to do it if we have trades
+    buy_asset = symbol_to_asset_or_token('LOLZ2')
+    buy_amount = deserialize_asset_amount(1)
+    sold_asset = symbol_to_asset_or_token('LOLZ')
+    sold_amount = deserialize_asset_amount(2)
+    rate = Price(buy_amount / sold_amount)
+    trade = Trade(
+        timestamp=Timestamp(12312312),
+        location=Location.BLOCKFI,
+        base_asset=buy_asset,
+        quote_asset=sold_asset,
+        trade_type=TradeType.BUY,
+        amount=buy_amount,
+        rate=rate,
+        fee=None,
+        fee_currency=None,
+        link='',
+        notes="",
+    )
+
+    database.add_trades([trade])
+    status, _ = GlobalDBHandler().rebuild_assets_list(database.conn)
+    assert status is False
+    # Now do it without the trade
+    database.delete_trade(trade.identifier)
+    status, _ = GlobalDBHandler().rebuild_assets_list(database.conn)
+    assert status
+    cursor = globaldb._conn.cursor()
+    # Check that the ethereum token was deleted
```

Comment:
Both here and in all your other "Check" comments. Delete them. Instead move them as the assertion error string. Example:

`assert r.fetchone() == (0,)` -> `assert r.fetchone() == (0,), 'Ethereum token should not have been deleted'`